deps/chakrashim/src/jsrtcontextshim.h

```diff
@@ -107,33 +107,35 @@ class ContextShim {
   IsolateShim * isolateShim;
   JsContextRef context;
   bool initialized = false;
-  bool initializing = false;
   bool exposeGC;
   JsValueRef keepAliveObject;
   int builtInCount;
   JsValueRef globalObjectTemplateInstance;
 
-  JsValueRef trueRef;
-  JsValueRef falseRef;
-  JsValueRef undefinedRef;
-  JsValueRef nullRef;
-  JsValueRef zero;
-  JsValueRef globalConstructor[GlobalType::_TypeCount];
-  JsValueRef globalObject;
-  JsValueRef proxyOfGlobal;
-  JsValueRef reflectObject;
-  JsValueRef reflectFunctions[ProxyTraps::TrapCount];
-
-  JsValueRef globalPrototypeFunction[GlobalPrototypeFunction::_FunctionCount];
+  JsValueRef trueRef = JS_INVALID_REFERENCE;
```

Comment:
C++ 11 relaxes the rule. See [this](http://stackoverflow.com/questions/13662441/c11-allows-in-class-initialization-of-non-static-and-non-const-members-what-c) and [this](https://msdn.microsoft.com/en-us/library/hh567368.aspx#corelanguagetable).